### DISCUSSION TITLE
zinit update --parallel leaves some unfinished background process

### DISCUSSION THREAD

**Discussion by OmeletWithoutEgg** [zdharma-continuum/zinit#499] [Q&A] [answered]
There is a option to update the plugins parallelly. Is it normal that, after running `zinit update --parallel`, it will leave some uncleaned child process? One can run `fg` or just run the parallel update in a shell inside another shell and exit.

**Comment by vladdoster** [comment]
Could you share screenshot(s) or video?

**Reply by OmeletWithoutEgg** [reply]
![image](https://user-images.githubusercontent.com/50587059/232233669-eb686e43-1faf-4be8-b5e8-4f5d8d754b3a.png)

The update can be done only after I commented https://github.com/zdharma-continuum/zinit/blob/main/zinit.zsh#L2949 (with the master branch)

Just noticed that I can't update anymore and found this issue https://github.com/zdharma-continuum/zinit/issues/502

Is the problem being solved in `fix/zinit-update-or-status` branch? It seems that `jobs -l` will have no uncleaned process in that branch.

**Reply by vladdoster** [reply, marked as answer]
@OmeletWithoutEgg,

Yes, it is being worked on. I plan on removing the current multi-processing logic in favor of using builtin `zargs` command. Using `zargs` should remove the issue you see regarding zombie jobs.

Feel free to try out the branch and give any feedback you might have (positive or negative).

I expect to have the refactor done sometime this weekend.

**Reply by OmeletWithoutEgg** [reply]
Hi, just wondering that will that branch eventually being merged to main?

**Reply by ErnestDong** [reply]
any updates now?
